packages/api/src/external/carequality/document/process-ihe-to-gw-document-query.ts

```diff
@@ -0,0 +1,189 @@
+import { out } from "@metriport/core/util/log";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { makeLambdaClient } from "@metriport/core/external/aws/lambda";
+import { errorToString } from "@metriport/core/util/error/shared";
+import { capture } from "@metriport/core/util/notifications";
+import { DocumentReference, DocumentQueryRespFromExternalGW } from "@metriport/ihe-gateway-sdk";
+import { isConvertible } from "../../fhir-converter/converter";
+import { Config } from "../../../shared/config";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { mapDocRefToMetriport } from "../../../shared/external";
+import { DocumentWithMetriportId } from "./shared";
+import { getNonExistentDocRefs } from "./get-non-existent-doc-refs";
+import { makeIheGatewayAPI } from "../api";
+import { createCQDocumentRetrievalRequests } from "./document-query-retrieval";
+import { getPatientWithDependencies } from "../../../command/medical/patient/get-patient";
+import { combineDocRefs } from "./shared";
+import { upsertDocumentToFHIRServer } from "../../fhir/document/save-document-reference";
+import { cqToFHIR } from "./shared";
+import { setDocQueryProgress } from "../../hie/set-doc-query-progress";
+import { processAsyncError } from "../../../errors";
+
+const region = Config.getAWSRegion();
+const iheGateway = makeIheGatewayAPI();
+const lambdaClient = makeLambdaClient(region);
+const lambdaName = Config.getIHEToExternalGwDocumentRetrievalsLambdaName();
+const parallelUpsertsToFhir = 500;
```

Comment:
500 seems like a lot to send to the FHIR server in parallel

Reply:
Ill move it to like 100 if that seems more reasonable